### PR TITLE
[custom-headers-in-response] - Added custom headers in response

### PR DESCRIPTION
**Pull request by PieroValdebenito**
. Added custom http.header as headers for http response
. Backward compatible

### CODE REVIEW

**Review by djaque — APPROVED** [review]
LGTM

File: http.go [hunk 1]
```diff
@@ -14,8 +14,9 @@ import (
 
 // Response is a struct to generate a response from POST/PUT requests
 type Response struct {
-	Code int
-	Body interface{}
+	Code    int
+	Body    interface{}
+	Headers map[string]string
```
Comment (mdiazv) [inline http.go]: The header order is important under certain special cases. In particular, when issuing the same header multiple times (which is valid, but not supported on this implementation).
https://stackoverflow.com/questions/750330/does-the-order-of-headers-in-an-http-response-ever-matter

We need to, at least, document that we don't support multiple headers and that the order is unspecified.

File: http.go [hunk 2]
```diff
@@ -27,6 +28,9 @@ type ErrorType struct {
 // WriteJSONResponse write to te response stream
 func WriteJSONResponse(w http.ResponseWriter, response *Response) {
 	w.Header().Set("Content-Type", "application/json")
+	for header, value := range response.Headers {
+		w.Header().Set(header, value)
```
Comment (mdiazv) [inline http.go]: Saving the result of `w.Header()` before the first call would save lots of function calls

File: http.go (hunk 1 above)
Comment (PieroValdebenito) [reply http.go]: Added comments

File: http.go [hunk 3]
```diff
@@ -26,6 +28,11 @@ type ErrorType struct {
 
 // WriteJSONResponse write to te response stream
 func WriteJSONResponse(w http.ResponseWriter, response *Response) {
+	for header, values := range response.Headers {
+		for i := range values {
+			w.Header().Add(header, values[i])
```
Comment (mdiazv) [inline http.go]: Array indirection may be expensive.
`for _, v := range values` should be quite faster.

Comment (mdiazv) [reply http.go]: Also, saving the repeated function call for w.Header() should squeeze a bit of extra performance.

**Review by mdiazv — COMMENTED** [review]
Looks great. A little bit of performance related comments though. As this code will be called on every request, should run as fast as possible.

**Review by mdiazv — APPROVED** [review]
👍

**Pull request merged**